recsys/recsys_tutorial.py

```diff
@@ -164,11 +187,11 @@ def polarity_negative(x):
 # ### Applying labeling functions to the training set
 #
 # We apply the labeling functions to the training set, and then filter out examples unlabeled by any LF, and combine the rest with the dev set to form our final training set.
+# We choose to train on our dev set in this case because it's relatively large (85.3k examples) and we didn't overfit to it when writing our LFs.
```

Comment:
In general, it should always be ok to train on the dev set. Even if you overfit, as long as the final labels are correct, the end model should benefit from it.
Note that we are also using dev set for training (along with labeled part of the train set) in the spouse tutorial.